cmd/skopeo/utils_test.go

```diff
@@ -197,6 +197,47 @@ func TestImageDestOptionsNewSystemContext(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestImageOptionsUsernamePassword(t *testing.T) {
```

Comment:
Non-blocking: Could this be turned into a table-driven test to reduce repetition?

Something approximately like
```go
for _, c := range []struct{
    args []string
    expected *types.DockerAuthConfig // data to expect, or nil if an error is expected
} {
    // test cases
} {
    opts := fakeImageDestOptions(t, "dest-", true, []string{}, c.args)
    res, err := opts.newSystemContext()
    if c.expected == nil {
        assert.Error(t, err)
    } else {
        require.NoError(t, err)
 	assert.Equal(t, &types.SystemContext{
 		DockerRegistryUserAgent: defaultUserAgent,
 		DockerAuthConfig:        c.expected,
 	}, res)
    }
}
```